src/zenml/cli/pipeline.py

```diff
@@ -87,25 +89,165 @@ def register_pipeline(source: str) -> None:
     pipeline_instance.register()
 
 
-@pipeline.command("run", help="Run a pipeline with the given configuration.")
+@pipeline.command("build", help="Build Docker images for a pipeline.")
+@click.argument("pipeline_name_or_id")
+@click.option(
+    "--version",
+    "-v",
+    type=str,
+    required=False,
+)
+@click.option(
+    "--config",
+    "-c",
+    "config_path",
+    type=click.Path(exists=True, dir_okay=False),
+    required=False,
+)
+@click.option(
+    "--stack",
+    "-s",
+    "stack_name_or_id",
+    type=str,
+    required=False,
+)
+@click.option(
+    "--output",
+    "-o",
+    "output_path",
+    type=click.Path(exists=False, dir_okay=False),
+    required=False,
+)
+def build_pipeline(
+    pipeline_name_or_id: str,
+    version: Optional[str] = None,
+    config_path: Optional[str] = None,
+    stack_name_or_id: Optional[str] = None,
+    output_path: Optional[str] = None,
+) -> None:
+    """Build Docker images for a pipeline.
+
+    Args:
+        pipeline_name_or_id: Name or ID of the pipeline.
+        version: Version of the pipeline.
+        config_path: Path to pipeline configuration file.
+        stack_name_or_id: Name or ID of the stack for which the images should
+            be built.
+        output_path: Optional file path to write the output to.
+    """
+    cli_utils.print_active_config()
+
+    if not Client().root:
+        cli_utils.error(
+            "`zenml pipeline build` can only be called within a ZenML "
+            "repository. Run `zenml init` at your source code root and try "
+            "again."
+        )
+
+    try:
+        id_ = UUID(pipeline_name_or_id, version=4)
+        pipeline_model = Client().get_pipeline(id=id_)
+    except ValueError:
+        pipeline_model = Client().get_pipeline(
+            name=pipeline_name_or_id, version=version
+        )
+
+    with cli_utils.temporary_active_stack(stack_name_or_id=stack_name_or_id):
+        pipeline_instance = BasePipeline.from_model(pipeline_model)
+        build = pipeline_instance.build(config_path=config_path)
+
+    if build:
+        cli_utils.declare(f"Created pipeline build `{build.id}`.")
+
+        if output_path:
+            cli_utils.declare(
+                f"Writing pipeline build output to `{output_path}`."
+            )
+            with open(output_path, "w") as f:
+                f.write(build.yaml())
+    else:
+        cli_utils.declare("No docker builds required.")
+
+
+@pipeline.command("run", help="Run a pipeline.")
+@click.argument("pipeline_name_or_id")
+@click.option(
+    "--version",
```

Comment:
This is all because we don't have a code repository concept yet and the only code version that is visible to ZenML is the current one (but also probably a bug).
Should use the same warning as in the other case.

Reply:
Actually the `reusing ...` message is only logged when it's actually reusing a registered pipeline. I'm guessing that both version 1 and 2 existed in the database, and explicitly specifying version 1 actually loaded version 2 (= the current local code) and then reused that pipeline version.